integrations/influxdb_metrics_dashboard/influxdb_metrics_dashboard.go

```
@@ -55,15 +61,55 @@ func addTelegrafEC2InstanceToStack(stack awscdk.Stack, stackProps awscdk.StackPr
 
 	timestreamPolicy.AttachToRole(instanceRole)
 
-	// Split the comma separated list of uris and create a map of instance ids and uris
-	influxDBEndpointsArr := strings.Split(influxDBEndpoints, ",")
-	influxDBInstances := make(map[string]string)
-	for _, instanceEndpoint := range influxDBEndpointsArr {
-		influxDBInstances[strings.Split(strings.Split(instanceEndpoint, "https://")[1], "-")[0]] = instanceEndpoint + "/metrics"
+	ctx := context.Background()
+	var awsCredentials aws.CredentialsProvider
+	awsConfig, err := config.LoadDefaultConfig(ctx)
+	if err != nil {
+		log.Printf("Error loading AWS config: " + err.Error())
+		os.Exit(1)
 	}
+	awsCredentials = awsConfig.Credentials
+	svc := timestreaminfluxdb.New(timestreaminfluxdb.Options{
+		Credentials: awsCredentials,
+		Region:      *stack.Region(),
+	})
+	ec2svc := ec2.New(ec2.Options{
+		Credentials: awsCredentials,
+		Region:      *stack.Region(),
+	})
 
+	vpcId := ""
 	telegrafInputOutputConfig := ""
-	for instanceId, instanceUri := range influxDBInstances {
+	instanceEndpoint := ""
+
+	// Split the comma separated list of Ids
+	influxDBIdArr := strings.Split(influxDBIds, ",")
+	for _, instanceId := range influxDBIdArr {
+
+		influxDBInstance, err := svc.GetDbInstance(ctx, &timestreaminfluxdb.GetDbInstanceInput{
+			Identifier: &instanceId,
+		})
+		if err != nil {
+			log.Printf("Error describing InfluxDB instance: %s", err)
+			os.Exit(1)
```

if there is an error with a given instance, can't we just log the error and continue on instead of exiting? this might mean a single bad instance could disrupt the deployment

---

Yes if one of the instance IDs are not correct, the deployment will fail. I think this is better functionality than someone accidentally doing a typo in an ID and later being confused why the instance isn't being monitored.

---

If I'm deploying this dashboard for a fleet of 30 instances and one instance happens to be unreachable for whatever reason, I would expect the deployment to succeed and can know either from the deployment logs or from the dashboard itself which instances are being monitored/failed to be monitored. In other words, I think the metrics dashboard deployment should be decoupled from the actual monitored instances where possible, though I understand that we are currently deriving other important details from the instances themselves.

---

I think 30 instances might be quite the outlier for use-case with this dashboard. Especially for visual real estate on the dashboard itself. I would expect the standard use-case to be one or a few instances, as even large customers will use only one.

From a usability perspective, I think the experience is better if users don't have to look for logs to determine why their dashboard didn't appear or is missing an instance's metrics. In either case they would need to re-deploy the dashboard, but if we fail early the user has the problem pointed out to them from the CDK error message.

My preference would be fail early, though open to discussions on your preference.